dali/test/python/test_external_source.py

```diff
@@ -0,0 +1,252 @@
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+import nvidia.dali.backend as backend
+import numpy as np
+from nose.tools import assert_raises
+import random
+
+class TestIterator():
+    def __init__(self, n, batch_size, dims = [2], as_tensor = False):
+        self.batch_size = batch_size
+        self.dims = dims
+        self.n = n
+        self.as_tensor = as_tensor
+        self.i = 0
+
+    def __len__(self):
+        return self.n
+
+    def __iter__(self):
+        # return a copy, so that the iteration number doesn't collide
+        return TestIterator(self.n, self.batch_size, self.dims, self.as_tensor)
+
+    def __next__(self):
+        np.random.seed(12345 * self.i + 4321)
+        def generate(dim):
+            shape = np.random.randint(1, 10, [dim]).tolist()
+            if self.as_tensor:
+                return np.random.ranf([self.batch_size] + shape)
+            else:
+                return [np.random.ranf(shape) for _ in range(self.batch_size)]
+        if self.i < self.n:
+            self.i += 1
+            if isinstance(self.dims, (list, tuple)):
+                return [generate(d) for d in self.dims]
+            else:
+                return generate(self.dims)
+        else:
+            self.i = 0
+            raise StopIteration
+    next = __next__
+
+def check_output(outputs, ref_out, ref_is_list_of_outputs = None):
```

Comment:
Doesn't we have something like this already in test_utils? If not move it there.